### PR TITLE
Adding Pre-commit configuration for Black and Flake8

### PR DESCRIPTION
**Pull request by chethanuk**
# Pre-commit, Black, and Flake8 Updates

## Summary

Adding pre-commit hooks for Black and Flake8, as well as updating the project configuration files for Flake8 to ensure consistent code style and catch common errors before committing.

## Changes

- Added pre-commit hooks for Black and Flake8 to the Maze project.
- Updated the Flake8 configuration file to ignore certain directories and files.
- Fixed various Flake8 errors and warnings found during the update process.

# Tests

The updated pre-commit hooks were tested by running `pre-commit run --all-files`
The Flake8 configuration changes were also tested by running flake8 on the repository both before and after the updates were applied.

cc:

### CODE REVIEW

**Comment by wangxiaoyou1993** [comment]
Thanks for adding the pre-commit. This helps a lot! Will review the changes today!

**Comment by chethanuk** [comment]
> Will review the changes today!

Great, Once this is merged, I have another PR adding other pre-commit hooks ready I will raise that as well.

File: mage_ai/cli/main.py [hunk 1]
```diff
@@ -21,14 +21,14 @@ def list_commands(self, ctx: Context):
 
 
 @app.command()
-def init(
-    project_path: str = typer.Argument(..., help='path of the Mage project to be created.')
```
Comment (wangxiaoyou1993) [inline mage_ai/cli/main.py:25]: That's the syntax of the typer library. Let's not change them.

**Comment by wangxiaoyou1993** [comment]
Close old PRs

**Pull request closed without merging**